### ISSUE TITLE
Soft link

### ISSUE THREAD

**Issue by exe1023** [SytemProgramming2015FallatNTU/Announcement#51]
我想確認一下我對處理soft link的理解有沒有錯
如果cdir/slink是個soft link且是指到cdir外的某個地方，在sync時sdir/user/slink也只需要讓他指到cdir外的某個地方
如果cdir/slink是個soft link且指到cidr內的cdir/target，在sync時要reconstruct他讓他指到sdir/target
這樣的理解是對的嗎?

**Comment by cchao** [comment]
No, you only have to send the result of `readlink()` from the client side and create the soft link with `symlink()` on the server side. You don't need to worry about where it points to.

Please refer to [Requirement](https://github.com/SytemProgramming2015FallatNTU/Announcement/wiki/Homework-2#requirement).
> Note that you have to pay special attention on handling the changes on links. In the case of symbolic link, you need to update the value of link instead of updating the content it points to. Even if the file which the symbolic link points to is out of the local repository, or it is a bad link, or there is a loop in the symbolic links, we should do the synchronization anyway.

**Comment by littlehug** [comment]
"you need to update the value of link instead of updating the content it points to"
有點不太懂，想再確認一下助教說的意思是指改變該link指向的文件裡的內容嗎? 還是link改指向別的文件?
謝謝!

**Comment by cchao** [comment]
指向別的文件
